storage/country.cpp

```diff
@@ -196,87 +211,98 @@ class DoStoreCountriesTwoComponentMwms
   }
 };
 
-class DoStoreFile2InfoSingleMwms
+class StoreFile2InfoTwoComponentMwms : public StoreTwoComponentMwmInterface
 {
-  TMapping m_idsMapping;
+  TMappingOldMwm m_idsMapping;
   map<string, CountryInfo> & m_file2info;
 
 public:
-  DoStoreFile2InfoSingleMwms(map<string, CountryInfo> & file2info)
-    : m_file2info(file2info) {}
+  StoreFile2InfoTwoComponentMwms(map<string, CountryInfo> & file2info) : m_file2info(file2info) {}
 
-  Country * operator()(TCountryId const & id, uint32_t /* mapSize */, int /* depth */,
-                       TCountryId const & /* parent */)
+  void Insert(string const & id, uint32_t mapSize, uint32_t /* routingSize */, int /* depth */,
+              TCountryId const & /* parent */) override
   {
+    if (mapSize == 0)
+      return;
+
     CountryInfo info(id);
-    m_file2info[id] = move(info);
-    return nullptr;
+    m_file2info[id] = info;
   }
+};
+}  // namespace
 
-  void operator()(TCountryId const & oldId, TCountryId const & newId)
+void LoadGroupTwoComponentMwmsImpl(int depth, json_t * node, TCountryId const & parent,
+                                   StoreTwoComponentMwmInterface & store)
+{
+  // @TODO(bykoianko) After we stop supporting two component mwms (with routing files)
+  // remove code below.
+  TCountryId file;
+  my::FromJSONObjectOptionalField(node, "f", file);
+  if (file.empty())
+    my::FromJSONObject(node, "n", file);  // If file is empty, it's the same as the name.
+
+  // We expect that mwm and routing files should be less than 2GB.
+  json_int_t mwmSize, routingSize;
+  my::FromJSONObjectOptionalField(node, "s", mwmSize);
+  my::FromJSONObjectOptionalField(node, "rs", routingSize);
+  ASSERT_LESS_OR_EQUAL(0, mwmSize, ());
+  ASSERT_LESS_OR_EQUAL(0, routingSize, ());
+
+  store.Insert(file, static_cast<uint32_t>(mwmSize), static_cast<uint32_t>(routingSize), depth,
+               parent);
+
+  json_t * children = json_object_get(node, "g");
+  if (children)
```

Comment:
In this case I think it's enough to implement following:
```c++
void FromJSON(json_t * root, json_t **value) { *value = root; }
```
And FromJSON for vector of json_t * will work nice, right?